src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1784,6 +1784,18 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 				} else if (methodReturnValue is SystemTypeValue) {
 					// SystemTypeValue can fullfill any requirement, so it's always valid
 					// The requirements will be applied at the point where it's consumed (passed as a method parameter, set as field value, returned from the method)
+				} else if (methodReturnValue is MergePointValue mergePointValue) {
```

Comment:
I think we can unify this with the if checks above:

Starting a line 1780:

```csharp
// Validate that the return value has the correct annotations as per the method return value annotations
if (returnValueDynamicallyAccessedMemberTypes != 0) {
    foreach (var uniqueValue in methodReturnValue.UniqueValues () {
      // put lines 1781 - 1789 of the original file here, but replace methodReturnValue with uniqueValue
    }
}
```

That way we don't need to duplicate the checks. Calling UniqueValues on null or on somehting that is not mergepoint is fine.